### PR TITLE
Move createUser validation from the mutation to the model’s method

### PR DESCRIPTION
**Pull request by Tom-Bonnike**
Title ☝️.
- 1f8f95449d0279b485682192b6de5acbd611a185 moves the validation itself.
- 1427c0ea5a338012a9fe1ad8de71dff8ae9db742 moves the tests.

In the mutation, we now only need two tests:
- On error, the errors array should be filled.
- On success, the new user should be returned and the errors array should be empty.

### CODE REVIEW

File: flow-typed/npm/pg_v7.x.x.js [hunk 1]
```diff
@@ -1,330 +0,0 @@
-// flow-typed signature: 719b210af83276ea2bbc0e9d0a84f2de
```
Comment (Tom-Bonnike) [inline flow-typed/npm/pg_v7.x.x.js]: Was causing issues on commit… hopefully resolved with its next version

File: src/models/User/createUser/integration.spec.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    const { user } = await createUser(validUser)
+    const countAfter = await database('users').count()
+
+    expect(countBefore[0].count).toEqual('0')
+    expect(countAfter[0].count).toEqual('1')
+    expect(user).toMatchObject({
+      id: 1,
+      name: 'Foo',
+      email: 'foo@bar.com'
+    })
```
Comment (Tom-Bonnike) [inline src/models/User/createUser/integration.spec.js]: Let‘s add a `expect(errors.length).toEqual(0)` assertion.

File: src/api/User/mutations/createUser/integration.spec.js [hunk 3]
```diff
@@ -1,144 +1,38 @@
 import gql from 'graphql-tag'
 import testClient from '@tests/client'
 import database from '@server/database'
-import {
-  INVALID_EMAIL_ERROR,
-  EMAIL_ALREADY_IN_USE_ERROR
-} from './helpers/validateEmail'
+
+const mutation = gql`
```
Comment (Tom-Bonnike) [inline src/api/User/mutations/createUser/integration.spec.js]: We could drop the `gql` dependency and use a `.graphql` file instead but that doesn’t change much 😄

Comment (PauuloG) [reply src/api/User/mutations/createUser/integration.spec.js]: Writing inline GraphQL should be avoided, especially if we modify this specific type. Then again tests won't pass if this definition is wrong, so it's pretty much the same

**Pull request merged**